src/components/ChatEntry.js

```diff
@@ -1,22 +1,38 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
+//spread 
+//TimeStamp has time prop
+const ChatEntry = ({ updateChat, ...message}) => {
+
+  const onLikeClick = () => {
+    updateChat({ ...message, liked: !message.liked });
+  };
+
 
-const ChatEntry = (props) => {
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{message.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{message.body}</p>
+        <p className="entry-time"><TimeStamp time={message.timeStamp} /> 
+        </p>
+        <button className="like" onClick={onLikeClick}>{message.liked ? '❤️':'🤍'}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
```

Comment:
Nice prop types. We should also have one for `id`. You're not currently explicitly using the id (it _is_ included in the copied message data ultimately used by the callback function), and it's being passed in by ChatLog, so it should be listed here.

Ideally, we _would_ probably mark all of these props as required (as you have), but since there was some test weirdness with the props due to how the test data was provided, we could have left some of them not required.

In general, if a property _isn't_ mark required, we should also ensure that our component works properly in its absence. For example, if the callback isn't supplied, we might avoid calling it, or even avoid rendering the heart as clickable at all. Or if the liked value isn't supplied, we might render the message without the liked indicator at all.